src/credentials/sso/cache.ts

```diff
@@ -31,32 +32,40 @@ export interface SsoCache {
 
 const cacheDir = join(homedir(), '.aws', 'sso', 'cache')
 
-export function getCache(directory = cacheDir): SsoCache {
+export function getCache(directory = cacheDir, statFunc = getFileStats): SsoCache {
+    try {
+        deleteOldFiles(directory, statFunc)
```

Comment:
how often is `getCache` called? will iterating all the files cause a lot of unnecessary filesystem activity?

Reply:
`deleteOldFiles` will have some impact on performance though it shouldn't be massive. Could add a module-scoped flag to skip if set. And then only set it when not running as a test.

Reply:
Also we will be deleting this functionality in the future

Reply:
Yeah, wonder if `getCache` itself is hot enough to warrant an optimization.  Out of scope for this PR, though.